source/common/stats/stats_impl.cc

```diff
@@ -135,35 +135,36 @@ bool TagExtractorImpl::extractTag(const std::string& stat_name, std::vector<Tag>
   return false;
 }
 
-RawStatData* HeapRawStatDataAllocator::alloc(const std::string& name) {
-  uint64_t num_bytes_to_allocate = RawStatData::structSize(name.size());
-  RawStatData* data = static_cast<RawStatData*>(::calloc(num_bytes_to_allocate, 1));
-  if (data == nullptr) {
-    throw std::bad_alloc();
-  }
-  data->checkAndInit(name, num_bytes_to_allocate);
+HeapStatData::HeapStatData(absl::string_view key) : name_(key.data(), key.size()) {}
 
+HeapStatData* HeapStatDataAllocator::alloc(absl::string_view name) {
+  // Do not truncate here for non-hot-restart case; truncate at the call-site in
```

Comment:
This is probably a weird nit, but I think this comment is adding confusion.  If we want to delegate truncation to the callsite, I think we can just make the comment (or something similar) since the truncation logic isn't really relevant here:
```
// Any expected truncation of name is done at the callsite. No truncation is
// required to use this allocator.
```

Reply:
done